### PR TITLE
Fix(Next-Gen CAREamics): ensure that the normalization stats are saved in the hyperparameters

### PR DESCRIPTION
**Pull request by jdeschamps**
## Description

> [!NOTE]  
> **tldr**: Currently, the data config saved in the checkpoint does not contain the normalization statistics.

When loading a checkpoint for prediction, statistics are systematically recalculated. The reason is that the data module saves the data configuration in the hyperparameters before the normalization statistics are computed.

This PR refactors the saving of the hyperparameters in a private method of the data module, and call it in `setup` after the training dataset has been created, ensuring the presence of the statistics.

# What changed

In `CAREamicsDataModule`:
- `self.save_hyperparameters` call has been refactored to `self._save_hparams` method
- call to `self._save_hparams` happens now in `setup`
- assertion added to test to verify the presence of the stats

### CODE REVIEW

File: src/careamics/lightning/dataset_ng/data_module.py [hunk 1]
```diff
@@ -336,6 +322,28 @@ def _sampler(self, dataset: Literal["train", "val", "predict"]) -> Sampler | Non
             sampler = None
         return sampler
 
+    def _save_hparams(self) -> None:
+        """Save configuration in hyperparameters."""
+        self.save_hyperparameters(
+            {"data_config": self.config.model_dump(mode="json")},
+            ignore=[
```
Comment (melisande-c) [inline src/careamics/lightning/dataset_ng/data_module.py:329]: oops just noticed these ignored params weren't updated when we changed the signature of the data module initialisation

**Review by melisande-c — APPROVED** [review]

File: src/careamics/lightning/dataset_ng/data_module.py (hunk 1 above)
Comment (jdeschamps) [reply src/careamics/lightning/dataset_ng/data_module.py:329]: Oh good catch, I didn't even think about it when I refactored it.

Comment (jdeschamps) [reply src/careamics/lightning/dataset_ng/data_module.py:329]: Done in https://github.com/CAREamics/careamics/pull/868/commits/e07d86306e443a51cd77cfd09741ee80ef80d9a6
